### ISSUE TITLE
Timewarrior's Gentoo test-image

### ISSUE THREAD

**Issue by lauft** [GothenburgBitFactory/timewarrior#380]
TL;DR: I am deactivating the _Gentoo Linux_ Docker test-image due to its high maintenance cost.

---
Timewarrior aims to run on many platforms and has thus build up a large docker image zoo (see https://travis-ci.org/GothenburgBitFactory/timewarrior) to run its test-suite on:
* Arch Linux
* CentOS
* Debian
* Fedora
* Gentoo Linux
* openSUSE
* Ubuntu

Errors in a test run usually have one the following reasons
* error in the application (this is what we want to spot!)
* missing/faulty configuration (e.g. wrong time zone)
* missing/conflicting package
* infrastructure

of which the top ones are the whole reason of CI (spotting errors in the application) and the lower ones are more related to system maintenance. At best, the system is set up and from then, if any, you only get errors from the test suite. For most distributions, this holds pretty good, however _Gentoo Linux_ seems to be the oddball in this collection. 

I did a quick and dirty statistic on the builds for the `dev` branch on Travis, and of the last 144 runs, 68 were red.

_Gentoo_ was among them with 60 errors and 3 failures; for all the other distributions the numbers were around 8-ish for errors 20-ish for failures (see below).

This means _Gentoo_ has roughly twice as many red builds, and in stark contrast to the other distributions the overall portion of those were _errors_ on _Gentoo_'s side (i.e. package conflicts, infrastructure) and not _failures_ in Timewarrior's test suite.
(Even accounting for 5 builds on Gentoo that errored because Travis-CI aborted them did not turn this statistic in _Gentoo_'s favour...)

To me this means fixing a CI error is mostly fixing _Gentoo_-Problems and not fixing errors in Timewarrior and unfortunately _Gentoo_ is not an easy system to maintain. All in all this requires to much of my time and therefore I decided to deactivate the _Gentoo_ Docker image.

If anyone feels up for the task of dedicated longterm _Gentoo_ maintenance, please contact me.

---
Here are the numbers of the 144 runs (ids 3-299) on Travis I analyzed:

Distribution | Errors | Failures
---------------|--------|---------               
Gentoo Linux   |     60 |        3
Fedora 28      |      7 |       17
Fedora 29      |      5 |       21
Debian Stable  |      4 |       21
Debian Testing |      4 |       21
Ubuntu 16      |      3 |       22
Ubuntu 18      |      6 |       18
openSUSE 15    |      4 |       25
Arch Linux     |     11 |       18
MacOS 10.13    |      1 |       30

**Comment by janko-j1** [comment]
Hi! I'm a user maintaining this package for Gentoo. Would it be possible to have access to the errors/failures that you found so that I can look into it and/or file the relevant bugs in our bugzilla? And also would it be possible to include the test-suite in the release tarball so that we can do our own testing? I'll report any software-related bug that I find.

**Comment by lauft** [comment]
Hi @janko-j1!

You can take a look at the logs at Travis-CI (https://travis-ci.org/github/GothenburgBitFactory/timewarrior/builds) of the failing builds of the dev branch e.g. https://travis-ci.org/github/GothenburgBitFactory/timewarrior/jobs/722279241 or https://travis-ci.org/github/GothenburgBitFactory/timewarrior/jobs/685521206.

Including the test suite into the release tarball should be no problem.

**Comment by janko-j1** [comment]
Thanks! I've looked at some of the failing builds and it does indeed seem to be maintenance problems but no real Gentoo bugs though as far as I can tell. I do not know how Travis-CI works but anyway if the test suite is included in the release tarball then I can do the testing part on my hardware (which is also done by a Gentoo developer on other hardware before merging into the Gentoo tree). What do you think?

You'll include the test suite in next release?

**Comment by lauft** [comment]
@janko-j1 I would prefer to have continuous testing again, but - as said before - the Gentoo image proved to be to hard to maintain for me. If fixing/maintaining the Gentoo Docker image is not an option to you, let's try switching to your proposed solution, i.e. including the test suite into the release tarball and getting error reports from you and your peers.

> You'll include the test suite in next release?

Yes
